docs/theme/build.ts

```diff
@@ -0,0 +1,245 @@
+            // Used for application colors
+            return token.path.join('.');
+          }
+        },
+      },
+    },
+    formats: {
+      monaco: ({ dictionary, platform }) => {
+        const theme: MonacoTheme = {
+          base: platform.themeType === 'dark' ? 'vs-dark' : 'vs',
+          inherit: true,
+          rules: [],
+          colors: {},
+        };
+        // Filter out the design tokens we don't want to add to the
+        // 'colors' object. This includes core colors defined in tokens/core.json5
+        // and syntax tokens defined in tokens/syntax
+        dictionary.allTokens
+          .filter((token) => {
+            return !['color', 'syntax'].includes(token.path[0]);
+          })
+          .forEach((token) => {
+            // Add each token to the colors object, the name is generated by the custom
+            // transform defined above
+            theme.colors[token.name] = token.value;
+          });
+
+        // Map the syntax styles
+        theme.rules = dictionary.allTokens
+          .filter((token) => {
+            return token.path[0] === 'syntax';
+          })
+          .map((token) => ({
+            token: token.name,
+            foreground: token.value,
+            fontStyle: token.fontStyle,
+          }));
+
+        // Style Dictionary formats expect a string that will be then written to a file
+        return JSON.stringify(theme, null, 2);
+      },
+      vsCodeTheme: ({ dictionary, platform }) => {
+        // VSCode theme JSON files have this structure
+        const theme: VSCodeTheme = {
+          name: `Nu Disco ${platform.themeType}`,
+          type: platform.themeType,
+          colors: {},
+        };
+
+        // Filter out the design tokens we don't want to add to the
+        // 'colors' object. This includes core colors defined in tokens/core.json5
+        // and syntax tokens defined in tokens/syntax
+        dictionary.allTokens
+          .filter((token) => {
+            return !['color', 'syntax'].includes(token.path[0]);
+          })
+          .forEach((token) => {
+            // Add each token to the colors object, the name is generated by the custom
+            // transform defined above
+            theme.colors[token.name] = token.value;
+          });
+
+        // Map the syntax styles
+        theme.tokenColors = dictionary.allTokens
+          .filter((token) => {
+            return token.path[0] === 'syntax';
+          })
+          .map((token) => ({
+            scope: token.name,
+            settings: {
+              foreground: token.value,
+              fontStyle: token.fontStyle,
+            },
+          }));
+
+        // Style Dictionary formats expect a string that will be then written to a file
+        return JSON.stringify(theme, null, 2);
+      },
+    },
+    filters: {
+      css: (token) => {
+        return token.path[0] === 'color';
+      },
+    },
+  },
+});
+
+['dark', 'light'].forEach(async (mode) => {
```

Comment:
async within forEach is technically not supported and may result in buggy behavior, better to use a for of loop here.

Or, if you want to run things in parallel, use this pattern:

```js
// wrapper async function for putting stuff in sequence
const runSD = async (mode) => {
  const sdExt = await sd.extend({ ... });
  await sdExt.buildAllPlatforms();
};

// to run stuff in parallel
await Promise.all(['dark', 'light'].map(runSD));
```